### PR TITLE
feat: make ZWaveNode class also emit it's events via this.driver.emit

### PR DESCRIPTION
**Pull request by sdrsdr**
As described in  #4914 it will be very hand to be able to subscribe for all network events via the driver event emitter, and this PR enables this. It's just a suggestion, I don't fully understand the typing of the event emitters but my typescript is happy with what I've done. Any suggestions are highly welcomed :)

### CODE REVIEW

**Comment by AlCalzone** [comment]
Hey, just wanted to let you know - I haven't forgotten this, but I need to get a couple of things stabilized before QoL things like this can land.

**Comment by sdrsdr** [comment]
No rush 😃 I'm using the package from my repo thanks to yarn. I'm glad that you even consider this feature

**Comment by AlCalzone** [comment]
Sorry, totally lost track of this while focusing on getting the driver ready for certification.
Will take a look on monday.

**Comment by sdrsdr** [comment]
Ah don't worry, I might endup maintaining a fork for our own needs so you can totally drop ths if you don't like it

File: packages/serial/src/ZWaveSerialPortBase.ts [hunk 1]
```diff
@@ -76,8 +76,15 @@ export function isZWaveSerialPortImplementation(
 	);
 }
 
+type write_cb=(err:Error|null|undefined)=>void
+export interface DuplexAlike {
+	// (chunk: any, encoding?: BufferEncoding | undefined, cb?: ((error: Error | null | undefined) => void) | undefined): boolean; (chunk: any, cb?: ((error: Error | null | undefined) => void) | undefined): boolean;
+	write(chunk:Buffer,cb?:write_cb|undefined):boolean;
+} ;
```
Comment (AlCalzone) [inline packages/serial/src/ZWaveSerialPortBase.ts:83]: @sdrsdr can you explain why this is needed? What's your use case?

In any case this feels out of place in this PR, I'd do this separately.

**Review by AlCalzone — CHANGES_REQUESTED** [review]
I'll rework how you forward the events as I'm not quite happy with how it is done. I'll make sure to include your initial commit, so it gets attributed to you.

File: packages/serial/src/ZWaveSerialPortBase.ts (hunk 1 above)
Comment (sdrsdr) [reply packages/serial/src/ZWaveSerialPortBase.ts:83]: I'm not sure how this got here :( It's part of me experimenting with passing non full duplex complaint interface there. You see we're experimenting with WebSockets  an the ws library does not provide a duplex compatible interface so I was trying to use TS to filter down the actually used methods. Did not get me too far as it's casted multiple times to different interfaces. You can totally omit it. Sorry for the inconvenience

Comment (AlCalzone) [reply packages/serial/src/ZWaveSerialPortBase.ts:83]: ok, I'm happy to loosen the definitions here if it makes sense though. Just need to understand the use-case.

**Pull request closed without merging**